src/rust/engine/src/nodes/session_values.rs

```diff
@@ -13,7 +14,8 @@ pub struct SessionValues;
 
 impl SessionValues {
     pub(super) async fn run_node(self, context: Context) -> NodeResult<Value> {
-        Ok(Value::new(context.session.session_values()))
+        let session_values = Python::with_gil(|py| context.session.session_values(py));
```

Comment:
I'm not a fan of taking the GIL here. The PyO3 migration guide recommends using `Arc<Py<PyAny>>` which is almost exactly the Pants `Value` type. We should just store `Value` and bind it to a `Python` context when necessary. Will update this PR accordingly.

Reply:
Latest commits fix this issue.